lib/dslash_policy.cuh

```diff
@@ -1131,212 +1012,10 @@ struct DslashFactory {
-  if(pthread_create(&interiorThread, NULL, launchInteriorKernel, &interiorParam)){
-    errorQuda("pthread_create failed");
-  }
-#else // single CPU thread per MPI process
-  const int packIndex = Nstream-1;
-  for(int i=3; i>=0; i--){
-    if(!dslashParam.commDim[i]) continue;
-    for(int dir=1; dir>=0; dir--){
-      PROFILE(inSpinor->recvStart(dslash.Nface()/2, 2*i+dir, dagger), profile, QUDA_PROFILE_COMMS_START);
-    }
-  }
-#endif
-  bool pack = false;
-  for (int i=3; i>=0; i--) 
-    if (dslashParam.commDim[i] && (i!=3 || getKernelPackT() || getTwistPack())) 
-      { pack = true; break; }
-
-#ifdef PTHREADS
-  if (pack){
-    PROFILE(cudaStreamWaitEvent(streams[packIndex], dslashStart, 0), 
-            profile, QUDA_PROFILE_STREAM_WAIT_EVENT); 
-  }
-#endif
-
-  // Initialize pack from source spinor
-  PROFILE(inSpinor->pack(dslash.Nface()/2, 1-parity, dagger, packIndex, false, twist_a, twist_b),
-	  profile, QUDA_PROFILE_PACK_KERNEL);
-
-  if (pack) {
-    // Record the end of the packing
-    PROFILE(cudaEventRecord(packEnd[0], streams[packIndex]), 
-	    profile, QUDA_PROFILE_EVENT_RECORD);
-  }
-#ifndef GPU_COMMS
-  for(int i = 3; i >=0; i--){
-    if (!dslashParam.commDim[i]) continue;
-
-    for (int dir=1; dir>=0; dir--) {
-      cudaEvent_t &event = (i!=3 || getKernelPackT() || getTwistPack()) ? packEnd[0] : dslashStart;
-
-      PROFILE(cudaStreamWaitEvent(streams[2*i+dir], event, 0), 
-	      profile, QUDA_PROFILE_STREAM_WAIT_EVENT);
-
-      // Initialize host transfer from source spinor
-      PROFILE(inSpinor->gather(dslash.Nface()/2, dagger, 2*i+dir), profile, QUDA_PROFILE_GATHER);
-
-      // Record the end of the gathering
-      PROFILE(cudaEventRecord(gatherEnd[2*i+dir], streams[2*i+dir]), 
-	      profile, QUDA_PROFILE_EVENT_RECORD);
-    }
-  }
-#endif // GPU_COMMS
-
-#endif // MULTI_GPU
-
-#if (!defined MULTI_GPU) || (!defined PTHREADS)
-  PROFILE(dslash.apply(streams[Nstream-1]), profile, QUDA_PROFILE_DSLASH_KERNEL);
-#endif
-
-#ifdef MULTI_GPU 
-
-#ifdef PTHREADS
-  if(pthread_join(receiveThread, NULL)) errorQuda("pthread_join failed");
-#endif
-#ifdef GPU_COMMS
-  bool pack_event = false;
-  for (int i=3; i>=0; i--) {
-    if (!dslashParam.commDim[i]) continue;
-
-    if ((i!=3 || getKernelPackT() || getTwistPack()) && !pack_event) {
-      cudaEventSynchronize(packEnd[0]);
-      pack_event = true;
-    } else {
-      cudaEventSynchronize(dslashStart);
-    }
 
-    for (int dir=1; dir>=0; dir--) {	
-      PROFILE(inSpinor->sendStart(dslash.Nface()/2, 2*i+dir, dagger), profile, QUDA_PROFILE_COMMS_START);
-      inSpinor->commsQuery(dslash.Nface()/2, 2*i+dir, dagger); // do a comms query to ensure MPI has begun
-    }
-  }
-#endif
-#ifdef PTHREADS
-  bool interiorLaunched = false;
-#endif
-  int completeSum = 0;
-  while (completeSum < pattern.commDimTotal) {
-    for (int i=3; i>=0; i--) {
-      if (!dslashParam.commDim[i]) continue;
+#if 0
```

Comment:
Can we just delete the code that is disabled with the '#if 0'? If not we should have a comment here.